Builds/CMake/RippledCore.cmake

```diff
@@ -119,7 +119,6 @@ target_sources (xrpl_core PRIVATE
 add_library (Ripple::xrpl_core ALIAS xrpl_core)
 target_include_directories (xrpl_core
   PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/ripple>
     # this one is for beast/legacy files:
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/beast/extras>
```

Comment:
I think there was a misunderstanding. We should remove these two include directories that dig into the `ripple` include directory, because they let files include headers without prefixing them with `ripple`:

```cmake
    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/ripple>
    # this one is for beast/legacy files:
    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/beast/extras>
```

and leave the line that includes the directory _containing_ the `ripple` include directory:

```cmake
    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
```

Reply:
The `beast/legacy` directory structure needs to move under the `ripple` include directory if we are not installing a `beast` include directory (which I don't think we are). I'll take a closer look today.